R/virtualenv.R

```diff
@@ -198,31 +206,31 @@ virtualenv_config <- function() {
          call. = FALSE)
   }
 
-  # find system python binary
-  pyver <- ""
-  python <- python_unix_binary("python")
-  if (is.null(python)) {
-    # try for python3 if we are on linux
-    if (is_linux()) {
-      python <- python_unix_binary("python3")
-      if (is.null(python))
-        stop("Unable to locate Python on this system.", call. = FALSE)
-      pyver <- "3"
-    }
-  }
+  # use currently configured version of Python
+  config <- py_config()
+  python <- config$python
+
+  # detect Python 2 vs Python 3
+  version <- package_version(config$version)
+  pyver <- if (version < "3") "" else "3"
 
   # find required binaries
   pip <- python_unix_binary(paste0("pip", pyver))
```

Comment:
I'm wondering if all of our uses of `python_unix_binary()` now need to be replaced with the result of `py_config()` (because otherwise we might use a mismatched version of python and pip, or python and virtualenv).

Reply:
That's a good idea. Perhaps instead of finding the associated binary on the PATH we should just check whether importing the associated Python module succeeds?

Reply:
I think those aren't just tests though, they are attempts to actually locate the binary so that we can call it. Of course, we could actually change things so we don't call the binary anymore but rather do e.g `python -m pip ...`, that might be an improvement. In any case, I think we need to either find the binary by looking alongside `python` (assuming that's where pip and virtualenv always live) or we need to invoke them via the `-m` command line arg).

Reply:
Ahh, got it -- I understand now. I'll spend some more time with this PR and see if I can clean this up.